Sources/ActorSingletonPlugin/ActorSingletonPlugin.swift

```diff
@@ -22,48 +23,70 @@ import DistributedActorsConcurrencyHelpers
 /// singleton running in the cluster.
 ///
 /// An actor singleton may run on any node in the cluster. Use `ActorSingletonSettings.allocationStrategy` to control
-/// its allocation. On candidate nodes where the singleton might run, use `ClusterSystem.singleton.ref(type:name:props:behavior)`
-/// to define actor behavior. Otherwise, call `ClusterSystem.singleton.ref(type:name:)` to obtain a ref. The returned
-/// `_ActorRef` is in reality a proxy which handle situations where the singleton is shifted to different nodes.
+/// its allocation. On candidate nodes where the singleton might run, use `ClusterSystem.singleton.proxy(type:name:system:props:factory)`
```

Comment:
So "proxy" is the wrong name here; since this is the API where "we might end up hosting it" and I don't want to always talk about proxies in APIs...

I think we should strive for an API like:

```swift
let singleton: ClusterSingleton(..., actorSystem: system)
await singleton.get(SomeActor.self)
```

perhaps? 

We can think if we want the `SingletonBoss<DA>` or just `SingletonBoss` hm, first might be easier to manage and implement? less type casting <3

Reply:
The DA impl doesn't have `ActorSingleton` anymore so yes, we can rename this to `ClusterSingleton` I think. 

I also thought about removing manager/boss because I don't think we really need it...

Reply:
I renamed things in https://github.com/apple/swift-distributed-actors/pull/980/commits/bfc125a1ccb8f47a0ea1cd8d9a175f63c0cc8cb4. Hopefully it's cleaner now.

Reply:
Yup nicer